pkgs/by-name/ed/ed/package.nix

```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  fetchurl,
+  lzip,
+  runtimeShellPackage,
+  stdenv,
+  testers,
+}:
+
+# Note: this package is used for bootstrapping fetchurl, and thus cannot use
+# fetchpatch! All mutable patches (generated by GitHub or cgit) that are needed
+# here should be included directly in Nixpkgs as files; further, those patches
+# should be included in the directory (or subdirectories) where this package file
+# belongs, given that ed now follows RFC 140.
```

Comment:
Do we need to specify the patch location? Using the same directory as the package was the convention even before RFC 140. And I would assume most Nixpkgs contributors to be familiar with Nixpkgs conventions – if they are not, the CI will catch it anyway.

Reply:
Given that
- I have found some expressions that refer to files outside directory 
  - SDL and gettext, that I am trying to migrate
- Many people are not well used to read the manuals and announcements
  - Things like "Why I can't use `qt6.callPackage ../by-name/. . .`?" -  or worse,  "Have you tried `qt6.callPackage ../by-name/. . .`?" - are commonplace

I preferred to be redundant.

Reply:
> * I have found some expressions that refer to files outside directory
>       
>       * SDL and gettext, that I am trying to migrate

I think those are very special cases and we should look at them individually. 


I don't think we need to explicitly mention it, but it also can't hurt. Maybe we can slim down the comment. No strong feelings.

Reply:
This convention of "writing everything related to a package in the same directory" is not really enforced.
The only place it is enforced is the by-name hierarchy.

And I know how difficult is to convince people about following unwritten and non-enforceable conventions.

Reply:
> I preferred to be redundant.

Then should we also add these three lines of comments to all other packages in by-name?

Reply:
> > I preferred to be redundant.
> 
> Then should we also add these three lines of comments to all other packages in by-name?

How many of them can't use `fetchPatch` because they are used to bootstrap the fetchers?

Reply:
Reworded to not cite RFC 140